alacritty_terminal/src/clipboard.rs

```diff
@@ -18,35 +18,47 @@ use std::ffi::c_void;
 use log::{debug, warn};
 
 use copypasta::nop_clipboard::NopClipboardContext;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(all(not(any(target_os = "macos", windows)), feature = "wayland"))]
 use copypasta::wayland_clipboard;
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+#[cfg(all(not(any(target_os = "macos", windows)), feature = "x11"))]
 use copypasta::x11_clipboard::{Primary as X11SelectionClipboard, X11ClipboardContext};
-use copypasta::{ClipboardContext, ClipboardProvider};
+#[cfg(any(feature = "x11", target_os = "macos", windows))]
+use copypasta::ClipboardContext;
+use copypasta::ClipboardProvider;
 
 pub struct Clipboard {
     clipboard: Box<dyn ClipboardProvider>,
     selection: Option<Box<dyn ClipboardProvider>>,
 }
 
 impl Clipboard {
-    #[cfg(any(target_os = "macos", target_os = "windows"))]
+    #[cfg(any(target_os = "macos", windows))]
     pub fn new() -> Self {
         Self::default()
     }
 
-    #[cfg(not(any(target_os = "macos", target_os = "windows")))]
-    pub fn new(display: Option<*mut c_void>) -> Self {
-        if let Some(display) = display {
-            let (selection, clipboard) =
-                unsafe { wayland_clipboard::create_clipboards_from_external(display) };
-            return Self { clipboard: Box::new(clipboard), selection: Some(Box::new(selection)) };
+    #[cfg(not(any(target_os = "macos", windows)))]
+    pub fn new(_display: Option<*mut c_void>) -> Self {
+        #[cfg(feature = "wayland")]
+        {
+            if let Some(display) = _display {
+                let (selection, clipboard) =
+                    unsafe { wayland_clipboard::create_clipboards_from_external(display) };
+                return Self {
+                    clipboard: Box::new(clipboard),
+                    selection: Some(Box::new(selection)),
+                };
+            }
         }
 
-        Self {
+        #[cfg(feature = "x11")]
+        return Self {
             clipboard: Box::new(ClipboardContext::new().unwrap()),
             selection: Some(Box::new(X11ClipboardContext::<X11SelectionClipboard>::new().unwrap())),
-        }
+        };
+
+        #[cfg(not(any(feature = "x11")))]
```

Comment:
I don't want to put words in @nixpulvis's mouth, but I think the question was why it was left as `any` when it's only checking for a single thing (`feature = "x11`). I tried this change locally and it appears to compile fine.

```suggestion
        #[cfg(not(feature = "x11"))]
```